src/webapp/components/dataset-periods/DataSetPeriodDates.tsx

```diff
@@ -0,0 +1,186 @@
+import React from "react";
+import styled from "styled-components";
+import { ConfirmationDialog, DatePicker } from "@eyeseetea/d2-ui-components";
+import { Typography } from "@material-ui/core";
+
+import { Id } from "$/domain/entities/Ref";
+import { useGetDataSetsByIds } from "$/webapp/hooks/useDataSets";
+import { addToDate } from "$/utils/date";
+import { PeriodDate, PeriodDetailsAttrs } from "$/domain/entities/PeriodDate";
+import i18n from "$/utils/i18n";
+
+export type DataSetPeriodDatesProps = {
+    dataSetIds: Id[];
+    onCancel: () => void;
+    onSave: (periodDate: PeriodDate) => void;
+};
+
+function useGetYears(props: { period: PeriodDate }) {
+    const { period } = props;
+    const { startDate, endDate, periods, years } = period;
+
+    const lastYear = years[years.length - 1];
+
+    const periodsByYear = React.useMemo(() => {
+        if (!startDate || !endDate) return [];
+
+        return years.map((year): PeriodDate["periods"][number] => {
+            const month = 4;
+            const day = 1;
```

Comment:
I guess this hardcoded 01/04 has some meaning for the domain, so we can move it to the entity with a nice name. Concepts that matter in the domain should not live solely in the presentation layer.

Reply:
These values are in the domain in [another branch](https://github.com/EyeSeeTea/dataset-configuration/blob/86859538bb83bf6eb103e5f98658e7eb61c83641/src/domain/entities/AppSettings.ts#L32) when we're implementing the app settings form.